### PR TITLE
S16 cart right

### PR DESCRIPTION
**Pull request by lingsat**

### CODE REVIEW

File: src/common/components/Count/Count.tsx [hunk 1]
```diff
@@ -42,37 +45,46 @@ const Count: FC<CountProps> = ({
   };
 
   const increaseCount = () => {
-    setCount((prev) => prev + 1);
+    const newCount = count + 1;
+    const newValidCount = getValidNumber(newCount.toString(), maxCount);
+    handleChangeAmount(newValidCount);
+    newCount > maxCount ? setError(true) : setError(false);
```
Comment (Yuliia143) [inline src/common/components/Count/Count.tsx]: Simplify

File: src/common/components/Count/Count.tsx [hunk 2]
```diff
@@ -42,37 +45,46 @@ const Count: FC<CountProps> = ({
...
   };
 
   const decreaseCount = () => {
-    setCount((prev) => prev - 1);
+    const newCount = count - 1;
+    const newValidCount = getValidNumber(newCount.toString(), maxCount);
+    handleChangeAmount(newValidCount);
+    newCount < ECount.MIN_COUNT_VALUE ? setError(true) : setError(false);
```
Comment (Yuliia143) [inline src/common/components/Count/Count.tsx]: x2

File: src/common/components/Count/Count.tsx [hunk 3]
```diff
@@ -42,37 +45,46 @@ const Count: FC<CountProps> = ({
...
   };
 
   const handleChangeCount = (event: ChangeEvent<HTMLInputElement>) => {
-    if (+event.target.value > maxCountValue) {
-      setMaxError(true);
+    if (+event.target.value > maxCount) {
+      setError(true);
+    } else {
+      setError(false);
```
Comment (Yuliia143) [inline src/common/components/Count/Count.tsx]: x3

File: src/common/components/Stars/Stars.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import star from "@Images/star.svg";
+import goldStar from "@Images/star_checked_gold.svg";
+import blackStar from "@Images/star_checked.svg";
+
+import "./Stars.scss";
+
+export enum EStarsColor {
+  BLACK = "blackStar",
+  GOLD = "goldStar",
+}
```
Comment (Yuliia143) [inline src/common/components/Stars/Stars.tsx]: Shouldn't be here if you export it

File: src/features/cart/components/CartItem/CartItem.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  };
+
+  const handleChangeCategory = (newCategory: string) => {
+    setNextCategory(newCategory);
+    if (invalidCategories.includes(newCategory)) {
+      const nextAmount = getNewCountAmount(cart, product.id, newCategory);
+      let amountSum = nextAmount + count.amount;
+
+      if (amountSum > product.stock[newCategory]) {
+        amountSum = product.stock[newCategory];
```
Comment (Yuliia143) [inline src/features/cart/components/CartItem/CartItem.tsx]: To avoid duplicates of product.stock[newCategory] you could create a variable

**Review by Yuliia143 — CHANGES_REQUESTED** [review]

File: src/common/components/Count/Count.tsx (hunk 1 above)
Comment (lingsat) [reply src/common/components/Count/Count.tsx]: Fixed - create single function for increment and decrement

File: src/common/components/Count/Count.tsx (hunk 2 above)
Comment (lingsat) [reply src/common/components/Count/Count.tsx]: Fixed - create single function for increment and decrement

File: src/common/components/Count/Count.tsx (hunk 3 above)
Comment (lingsat) [reply src/common/components/Count/Count.tsx]: Fixed - replace with setError(+event.target.value > maxCount);

File: src/common/components/Stars/Stars.tsx (hunk 4 above)
Comment (lingsat) [reply src/common/components/Stars/Stars.tsx]: Fixed - replace to common types. Also, did the same for common Button and Count components

File: src/features/cart/components/CartItem/CartItem.tsx (hunk 5 above)
Comment (lingsat) [reply src/features/cart/components/CartItem/CartItem.tsx]: Fixed - create a variable

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**